cns/restserver/internalapi.go

```diff
@@ -254,3 +262,57 @@ func (service *HTTPRestService) CreateOrUpdateNetworkContainerInternal(req cns.C
 	return returnCode
 
 }
+
+// Try to register node with DNC when CNS is started in managed DNC mode
+func RegisterNode(httpRestService cns.HTTPService, dncEP, infraVnet, nodeID string) {
+	logger.Printf("[Azure CNS] Registering node %s with Infrastructure Network: %s PrivateEndpoint: %s", nodeID, infraVnet, dncEP)
+
+	var (
+		numCPU              = runtime.NumCPU()
+		url                 = fmt.Sprintf(common.RegisterNodeURLFmt, dncEP, infraVnet, nodeID, dncApiVersion)
+		response            *http.Response
+		err                 = fmt.Errorf("")
+		body                bytes.Buffer
+		httpc               = common.GetHttpClient()
+		nodeRegisterRequest cns.NodeRegisterRequest
+	)
+
+	nodeRegisterRequest.NumCPU = numCPU
+	supportedApis, msg := nmagentclient.GetNmAgentSupportedApis("")
+
+	if msg != "" {
+		logger.Printf("[Azure CNS] Failed to retrieve SupportedApis from NMagent of node %s with Infrastructure Network: %s PrivateEndpoint: %s",
+			nodeID, infraVnet, dncEP)
+	}
+
+	nodeRegisterRequest.NmAgentSupportedApis = supportedApis
+	if err := json.NewEncoder(&body).Encode(nodeRegisterRequest); err != nil {
+		log.Errorf("encoding json failed with %v", err)
+		return
+	}
+
+	for sleep := true; err != nil; sleep = true {
```

Comment:
I know this is just moved code :) but I think we can use ticker to avoid sleeps and returning an error after a certain amount of time
```
ticker := time.NewTicker(time.Second * 5)
for {
  select {
  case <-ticker.C:
  	response, err = httpc.Post(url, "application/json", &body)
		if err == nil {
...
...
  case <-time.After(5 * time.Minute):
  	logger.errorf("no response from nmAgent after an hour") 
        return...
  }
}
```

Reply:
Agree with Mat